src/bindgen/cdecl.rs

```diff
@@ -131,7 +131,14 @@ impl CDecl {
                 self.build_type(t, false);
             }
             Type::Array(ref t, ref constant) => {
-                let len = constant.as_str().to_owned();
+                let mut len = constant.as_str().to_owned();
+                // Checks if a pointer represented as an array is const
+                let is_const = if let Some('c') = len.chars().last() {
```

Comment:
This looks a bit hacky :/

Reply:
Yep, it is :)

I didn't want to change a lot of code just to insert the `const` identifier, so since the length of an array is a `String`, I thought to implement this simple trick. The best solution would be adding a new type for `arrays-as-pointer`, but is it worth just for some annotations? It isn't in my opinion. If you have better ideas, I'm happy to listen to them

Reply:
I think this is not quite acceptable. There's nothing preventing the user to insert a `c` as an array length, right?

Your patch breaks this, for example:

```rust
pub const c: usize = 4;
pub type C = *const [i16; c];

#[no_mangle]
pub extern "C" fn foo(foo: C) {}
```